### PR TITLE
fix: JUnitReport.xml only contained 50 test results

### LINKED ISSUE #1634: JUnitReport.xml only includes 50 tests out of 73
👋 Thank you for this amazing tool.

**Describe the bug**
There are 2 issues that are probably related:
1. When I run tests via flank with `max-test-shards: "5"` for some reason it uses only 1 shard, I can't find the reason why.
2. The `JUnitReport.xml` which is created only contains 50 tests, while `matrix_0/Nexus5-23-en-portrait/test_result_1.xml` has 73 tests.

Is there any hard-coded value of the number of tests `JUnitReport.xml` can have?

**To Reproduce**
> Steps to reproduce the behaviour:

Run tests via flank with following configuration:

```yaml
gcloud:
  app: '/Users/ddanylyk/.../build/testlab/dummy.apk'
  test: '/Users/ddanylyk/Work/.../debug/android-debug-androidTest.apk'
  use-orchestrator: false
  record-video: true
  auto-google-login: false
  environment-variables:
    coverage: false
    clearPackageData: false
    coverageFile: /sdcard/coverage.ec
  device:
    - model: Nexus5
      version: 23
      locale: en
      orientation: portrait
  test-targets:
     - 'notAnnotation QuarantineTest'
flank:
  max-test-shards: '5'
  full-junit-result: true
  shard-time: '120'
  num-repeat-tests: '1'
  disable-sharding: 'false'
  smart-flank-gcs-path: gs://...testlabRunAndroidTest.xml
```

**Expected behaviour**
1. Flank should use more than 1 shard if `max-test-shards: '5'`.
2. `JUnitReport.xml` should contain all tests, not just 50.

**Details (please complete the following information):**
> Have you tested on the latest Flank snapshot?

No

> Post the output of ` flank --version`.

v20.09.3 (tried v21.02.0 same issue)

**Additional context**

Exanple of `matrix_0/Nexus5-23-en-portrait/test_result_1.xml` file:

```xml
<?xml version='1.0' encoding='UTF-8' ?>
<testsuite name="" tests="73" failures="0" errors="0" skipped="0" time="25.17" timestamp="2021-02-26T03:07:15" hostname="localhost">
  <properties />
  <testcase name="test1" classname="...CustomTagHandlerTest" time="0.306" />
  <testcase name="test2" classname="...CustomTagHandlerTest" time="0.306" />
...71 more entries
</testsuite>
```

Example of `JUnitReport.xml` file:

```xml
<?xml version='1.0' encoding='UTF-8' ?>
<testsuites>
  <testsuite name="Nexus5-23-en-portrait" tests="50" failures="0" flakes="0" errors="0" skipped="0" time="25.170" timestamp="2021-02-26T03:06:50" hostname="localhost">
    <testcase name="test1" classname="...CustomTagHandlerTest" time="0.489">
      <webLink>https://console.firebase.google.com/project/.../testcases/1</webLink>
    </testcase>
...49 more entries
</testsuites>

```

### PR DESCRIPTION
**Pull request by dmytrodanylyk**
Fixes #1634

Existing tests should not be broken

## Checklist

- [x] Documented
- [x] Unit tested
- [x] Integration tests updated

### CODE REVIEW

**Comment by dmytrodanylyk** [comment]
I have read the CLA Document and I hereby sign the CLA

File: test_runner/src/main/kotlin/ftl/gc/GcToolResults.kt [hunk 1]
```diff
@@ -123,7 +124,10 @@ object GcToolResults {
         .executeWithRetry()
 
     // Lists Test Cases attached to a Step
-    fun listTestCases(toolResultsStep: ToolResultsStep): ListTestCasesResponse {
+    fun listTestCases(
```
Comment (artour-bakiev) [inline test_runner/src/main/kotlin/ftl/gc/GcToolResults.kt:127]: Looks like we should update unit tests as well.

File: test_runner/src/main/kotlin/ftl/gc/GcToolResults.kt [hunk 2]
```diff
@@ -135,7 +139,19 @@ object GcToolResults {
...
+            .setPageToken(pageToken)
+            .executeWithRetry()
+    }
+
+    fun listAllTestCases(results: ToolResultsStep): List<TestCase> {
+        var response = listTestCases(results)
+        val testCases = response.testCases.toMutableList()
+        while (response.nextPageToken != null) {
+            response = listTestCases(results, response.nextPageToken)
+            testCases += response.testCases ?: emptyList()
```
[the remark this answers is not in the document]
Comment (dmytrodanylyk) [reply test_runner/src/main/kotlin/ftl/gc/GcToolResults.kt:152]: I am not sure, this code is just copy/paste from a similar code in this file (see `listAllEnvironments`, `listAllSteps`)

**Comment by pawelpasterz** [comment]
Does it solve the sharding problem as well?

**Comment by dmytrodanylyk** [comment]
> Does it solve the sharding problem as well?

Nope, just `JUniteResults.xml` fix.

**Comment by piotradamczyk5** [comment]
@Mergifyio rebase

**Review by piotradamczyk5 — APPROVED** [review]

**Review by Sloox — APPROVED** [review]

**Pull request merged**